low-level/files/opt/seagate/sspl/setup/resource_map/server.py

```diff
@@ -32,18 +31,100 @@ class ServerMap(ResourceMap):
 
     def __init__(self):
         """Initialize server"""
-        pass
+        super().__init__()
+        ipmi_factory = IpmiFactory()
+        self._ipmi = ipmi_factory.get_implementor("ipmitool")
+        self.platform_sensor_list = ['Temperature', 'Voltage', 'Current']
+        self.server_frus = {
+            'platform_sensors': self.get_platform_sensors,
+        }
 
     def get_health_info(self, rpath):
         """
         Get health information of fru in given resource id
 
         rpath: Resource id (Example: nodes[0]>compute[0]>hw>disks)
         """
-        pass
+        fru_found = False
+        info = dict()
+        nodes = rpath.strip().split(">")
+        leaf_node, _ = self.get_node_details(nodes[-1])
+        if leaf_node == "compute":
+            for fru in self.server_frus:
+                info.update({fru: self.server_frus[fru]()})
+            info["last_updated"] = int(time.time())
+            fru_found = True
+        else:
+            fru = None
+            for node in nodes:
+                fru, _ = self.get_node_details(node)
+                if self.server_frus.get(fru):
+                    fru_found = True
+                    info = self.server_frus[fru]()
+                    break
+        if not fru_found:
+            raise ResourceMapError(
+                errno.EINVAL,
+                "Health provider doesn't have support for'{rpath}'.")
+        return info
+
+    @staticmethod
+    def get_node_details(node):
+        """
+        Parse node information and returns left string and instance.
+        Example
+            "storage"    -> ("storage", "*")
+            "storage[0]" -> ("storage", "0")
+        """
+        res = re.search(r"(\w+)\[([\d]+)\]|(\w+)", node)
+        inst = res.groups()[1] if res.groups()[1] else "*"
+        node = res.groups()[0] if res.groups()[1] else res.groups()[2]
+        return node, inst
+
+    def get_platform_sensors(self):
+        sensor_list = ['Temperature', 'Voltage', 'Current']
+        data = self.build_platform_sensor_data(sensor_list)
+        return data
+
+    def get_json_formatted_reading(self, reading):
+        uid = '_'.join(reading[0].split())
+        status = reading[2]
```

Comment:
whats the value for 'status', which is being fed in health atlast

Reply:
value is "ok". updated sample output in function doc.

Reply:
Sensor has status ['ok', 'ns'].
so,
```
status = 'OK' if reading[2] == 'ok' else 'NOT OK'

Reply:
Done